### DISCUSSION TITLE
총구에서 총알이 나가는 걸 구현하려고 하는데...

### DISCUSSION THREAD

**Discussion by cinephile231** [naverz/zepeto-studio-kor#753] [5. Q&A]
총구에서 총알이 나가는 걸 구현하려고 하는데...
왜 총알이 자기 맘대로 나갈까요?

Physics.Raycast와 LineRender를 제어해서 구현하려고 했는데,
Component의 enabled가 제대로 작동을 안 하고 
LineReander도, Raycast도 제가 원하는 위치(총구의 z방향)로 나가지 않습니다.

그래서,
총구에서 총알이 생성되고
rigidbody의 AddForce로 구현을 해야겠다!
해서 만들었는데, 이걸 월드로 좌표를 찍어도, 로컬로 찍어도...
엉뚱하게 나갑니다. 

제페토의 Transform와 유니티의 Transform이 다른 건가요?
제가 뭘 놓치고 있는 건가요? 
이 좌표값의 레이턴시는 왜 생기는 건가요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. transform은 동일합니다.
캐릭터가 바라보는 방향대로 총알이 날아가는 예제 코드 안내 드립니다.
예제에서는 총알의 속도나 궤적은 Rigidbody로 설정하였으나, 방향 코드 부분만 참고하시면 좋을 것 같습니다. 


//총알 방향 설정
const dir = this.zepetoCharacter.gameObject.transform.forward;
//총알 프리팹 생성
var bullet = Object.Instantiate(this.bulletFactory) as GameObject;
//총알이 총구에서 나가도록 위치 세팅
bullet.transform.position = this.firePosition.transform.position;
//날아가는 파워 세팅
const power = 50;
//Bullet스크립트의 발사 함수 호출
bullet.GetComponent<BulletController>().Fire(dir, power);

<hr/>

//Bullet스크립트의 발사 함수 선언부

public Fire(dir:Vector3, power:number)
    {
        const x = dir.x * power;
        const y = dir.y * power * 1.5;
        const z = dir.z * power;
        const velocity = new Vector3(x,y,z);
        this.GetComponent<Rigidbody>().velocity = velocity;
    }
